fourcc_header_gen.c

```diff
@@ -0,0 +1,164 @@
+#include <stdio.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <drm_fourcc.h>
+#include "config.h"
+
+struct format
+{
+	char format[25];
+	uint32_t fourcc;
+};
+
+static struct format *formats;
+static unsigned int format_count;
+
+void get_fourcc(unsigned int j, size_t size, char *map)
```

Comment:
Why do we need to parse the fourcc code? We only want to generate this kind of code:

```c
switch (format) {
case DRM_FORMAT_XXX:
    return "XXX";
…
}
```

Reply:
You're right, this didn't occur to me until you brought it out.